binderhub/main.py

```diff
@@ -25,12 +25,9 @@ class ParameterizedMainHandler(BaseHandler):
     """Main handler that allows different parameter settings"""
 
     def get(self, provider_prefix, _unescaped_spec):
-        # re-extract spec from request.path
-        # get the original, raw spec, without tornado's unquoting
-        # this is needed because tornado converts 'foo%2Fbar/ref' to 'foo/bar/ref'
         prefix = '/v2/' + provider_prefix
-        idx = self.request.path.index(prefix)
-        spec = self.request.path[idx + len(prefix) + 1:]
+        spec = self.get_spec_from_request(prefix)
+        spec = spec.rstrip("/")
```

Comment:
`spec = spec.rstrip("/")` also fix for https://github.com/jupyterhub/binderhub/issues/622.

Reply:
that line also does the same for nbviewer as done for build in https://github.com/jupyterhub/binderhub/blob/master/binderhub/builder.py#L205